pkg/uhttp/wrapper.go

```diff
@@ -45,6 +45,9 @@ func NewBaseHttpClient(httpClient *http.Client) *BaseHttpClient {
 
 func WithJSONResponse(response interface{}) DoOption {
```

Comment:
Wonder if we should have a generic `WithResponse()` method that does the same as these, but switches on the content type.